lib/wherebot.rb

```diff
@@ -72,21 +72,15 @@ def body
 
     private
 
-    def headers
-      @headers ||= @imap.fetch(@id, "ENVELOPE")[0].attr["ENVELOPE"]
-    end
-
-    def full_body
-      @full_body ||= @imap.fetch(@id, 'BODY[TEXT]')[0].attr['BODY[TEXT]']
+    def mail
+      @mail ||= Mail.new(@imap.fetch(@id, 'BODY.PEEK[]')[0].attr['BODY[]'])
     end
 
     def get_text
-      body = full_body.split("Content-Type: text/")[1] || full_body
-      body = body.split("Content-Transfer-Encoding: ")[1] || body
-
-      if body.first(4) == "html"
-        body = ActionController::Base.helpers.strip_tags(body) || body
-      end
+      message = mail
+      message = message.parts[0] while message.multipart?
+      body = message.body.decoded
+      strip_tags(body) if message.content_type.try(:start_with?, "text/html")
```

Comment:
`strip_tags` creates a new string without altering the original, so you should either return here or do an assignment.

Reply:
Do we need this line, now that Mail is handling parts?